yt/yt/server/master/security_server/access_log.cpp

```diff
@@ -278,6 +280,27 @@ bool IsAccessLoggedType(const EObjectType type)
     return typesForAccessLog.contains(type);
 }
 
+bool IsAccessLogEnabled(NCellMaster::TBootstrap* bootstrap)
+{
+    if (!bootstrap->GetConfigManager()->GetConfig()->SecurityManager->EnableAccessLog) {
+        return false;
+    }
+
+    if (bootstrap->GetHydraFacade()->GetHydraManager()->IsRecovery()) {
+        return false;
+    }
+
+    if (!bootstrap->GetHydraFacade()->GetHydraManager()->IsLeader()) {
+        return true;
+    }
+
+    if (bootstrap->GetCellManager()->GetTotalPeerCount() <= 1) {
```

Comment:
I don't quite get it: if there's just one peer then it must be the leader. So why the if above is not taken then?

Anyway, it would be great to see a new test for this change.

Reply:
There is a negation above: we enable logging in case if the peer is not a leader. And also we enable logging here in case if there are no followers because only one peer exists.

Reply:
Indeed!